src/KafkaW/Producer.cpp

```diff
@@ -5,95 +5,17 @@ namespace KafkaW {
-
-void ProducerMsg::deliveryError() {}
-
-void Producer::deliveredCallback(rd_kafka_t *RK,
-                                 rd_kafka_message_t const *Message,
-                                 void *Opaque) {
-  auto Self = reinterpret_cast<Producer *>(Opaque);
-  if (!Message) {
-    LOG(Sev::Error, "IID: {}  ERROR msg should never be null", Self->id);
-    ++Self->Stats.produce_cb_fail;
-    return;
-  }
-  if (Message->err) {
-    LOG(Sev::Error, "IID: {}  ERROR on delivery, {}, topic {}, {} [{}] {}",
-        Self->id, rd_kafka_name(RK), rd_kafka_topic_name(Message->rkt),
-        rd_kafka_err2name(Message->err), Message->err,
-        rd_kafka_err2str(Message->err));
-    if (Message->err == RD_KAFKA_RESP_ERR__MSG_TIMED_OUT) {
-    }
-    if (auto &Callback = Self->on_delivery_failed) {
-      Callback(Message);
-    }
-    ++Self->Stats.produce_cb_fail;
-  } else {
-    if (auto &Callback = Self->on_delivery_ok) {
-      Callback(Message);
-    }
-
-    ++Self->Stats.produce_cb;
-  }
-}
-
-void Producer::errorCallback(rd_kafka_t *RK, int Err_i, char const *Message,
-                             void *Opaque) {
-  UNUSED_ARG(RK);
-  auto Self = reinterpret_cast<Producer *>(Opaque);
-  auto ERR = static_cast<rd_kafka_resp_err_t>(Err_i);
-  Sev Level = Sev::Warning;
-  if (ERR == RD_KAFKA_RESP_ERR__TRANSPORT) {
-    Level = Sev::Error;
-  } else {
-    if (Self->on_error) {
-      Self->on_error(Self, ERR);
-    }
-  }
-  LOG(Level, "Kafka cb_error id: {}  broker: {}  errno: {}  errorname: {}  "
-             "errorstring: {}  message: {}",
-      Self->id, Self->ProducerBrokerSettings.Address, Err_i,
-      rd_kafka_err2name(ERR), rd_kafka_err2str(ERR), Message);
-}
-
-int Producer::statsCallback(rd_kafka_t *RK, char *Json, size_t Json_len,
-                            void *Opaque) {
-  auto Self = reinterpret_cast<Producer *>(Opaque);
-  LOG(Sev::Debug, "IID: {}  INFO cb_stats {} length {}   {:.{}}", Self->id,
-      rd_kafka_name(RK), Json_len, Json, Json_len);
-  // What does librdkafka want us to return from this callback?
-  return 0;
-}
-
-void Producer::logCallback(rd_kafka_t const *RK, int Level, char const *Fac,
-                           char const *Buf) {
-  UNUSED_ARG(Level);
-  auto self = reinterpret_cast<Producer *>(rd_kafka_opaque(RK));
-  LOG(Sev::Debug, "IID: {}  {}  fac: {}", self->id, Buf, Fac);
-}
-
-void Producer::throttleCallback(rd_kafka_t *RK, char const *Name,
-                                int32_t Broker_id, int Throttle_time_ms,
-                                void *Opaque) {
-  UNUSED_ARG(RK);
-  auto Self = reinterpret_cast<Producer *>(Opaque);
-  LOG(Sev::Debug, "IID: {}  INFO cb_throttle  broker_id: {}  broker_name: {}  "
-                  "throttle_time_ms: {}",
-      Self->id, Broker_id, Name, Throttle_time_ms);
-}
-
 Producer::~Producer() {
   LOG(Sev::Debug, "~Producer");
-  if (RdKafkaPtr) {
+  if (ProducerPtr) {
     int TimeoutMS = 1;
     int OutQueueLength = 0;
     while (true) {
```

Comment:
I do not like this while statement and I do not like how timeouts are handled handled here. It should be a simple thing to write something that is easier to understand.